### PR TITLE
Codes revamp

### PR DESCRIPTION
**Pull request by SwiftieHusband**
image
image
image

Couldn't get it to where the badge for "NEW" didn't look tacky..
Included the new CMS ID so we don't have to bother changing the json when we merge this.

### CODE REVIEW

File: app/codes/page.tsx [hunk 1]
```diff
@@ -3,19 +3,22 @@
...
-      <div>Click to copy</div>
-      <div className="flex flex-row flex-wrap justify-center gap-2">
-        {codesContent.active.map((code: string) => (
-          <CopyButton key={code} label={code} />
+      <div className="grid md:grid-cols-2 lg:grid-cols-3 justify-center gap-2">
+        {codesContent.active.map((code: Code) => (
+          <CodeCard
+            key={code.code}
+            code={code}
+            rewards="test rewards"
```
Comment (KMontag42) [inline app/codes/page.tsx:20]: Looks like this needs to be updated

File: components/CodeCard.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+"use server";
+
+import { Card, CardContent } from "@/components/ui/card";
+import CopyButton from "./CopyButton";
```
Comment (KMontag42) [inline components/CodeCard.tsx:4]: Prefer to use @/components/CopyButton for import

File: app/codes/page.tsx [hunk 3]
```diff
@@ -3,19 +3,22 @@
 import Image from "next/image";
 
 import { getCodesCmsData } from "@/lib/server/cms-data";
-import { Button } from "@/components/ui/button";
-import CopyButton from "@/components/CopyButton";
+import { Code } from "@/lib/cms-types";
```
Comment (KMontag42) [inline app/codes/page.tsx:6]: You can import this as just the type since that’s all you need from it

**Review by KMontag42 — CHANGES_REQUESTED** [review]

File: app/codes/page.tsx (hunk 1 above)
Comment (SwiftieHusband) [reply app/codes/page.tsx]: Fixed, this was dead code.

File: components/CodeCard.tsx (hunk 2 above)
Comment (SwiftieHusband) [reply components/CodeCard.tsx]: good catch

File: app/codes/page.tsx (hunk 3 above)
Comment (SwiftieHusband) [reply app/codes/page.tsx]: Like this? `import type { Code } from "@/lib/cms-types";` Done.

**Review by KMontag42 — APPROVED** [review]
LGTM

**Pull request merged**